plugin/vimwatcher.py

```diff
@@ -7,7 +7,6 @@
 from aw_core.models import Event
 from aw_client.client import ActivityWatchClient
 
-logging.basicConfig(level=logging.INFO)
```

Comment:
This is the only line you should need to change.
The rest of this change seems odd, you are removing the logging completely when not in testing now (even logging to file).

The correct solution should be to move this row into main and set it to INFO if testing and WARNING otherwise.

Reply:
Hm yes, the commit is a bit confusing sorry. I think using basicConfig() and setup_logging() at the same time does not work very well together.

Just setting the loglevel in basicConfig() is not enough, because setup_logging() is also setting the loglevel. calling basicConfig() after setup_logging does also not work, because basicConfig() does not do anything if the root logger has already a handler attached (which is the case if setup_logging() is called with log_stderr=True before). Note that logging to a file is not enabled for testing=False anyway in the current situation.

Without touching setup_logging() I think the best solution would be to not use basicConfig() and simply reset the root level to warning after setup_logging(). Not using setup_logging() seems also like a valid approach. Using both at the same time and setting the log level to warning, I couldn't figure out any other approach then the one I suggested. It seemed to be the minimal invasive thing to do, but I agree, it is pretty confusing, but I don't know how to improve it. Do you have a suggestion or want to replace my commit with your own solution?

Reply:
Ah good point, that makes it kind of confusing.

Would it work if we changed the root logger right after calling setup_logging then?

    setup_logging(name, testing=testing, log_stderr=True, log_file=testing)
    root_logger = logging.getLogger()
    root_logger.setLevel(logging.INFO if testing else logging.warning)

> Note that logging to a file is not enabled for testing=False anyway in the current situation.

Maybe we should enable that though

Reply:
Sorry for the late reply, I kind of lost track of this issue.

> Would it work if we changed the root logger right after calling setup_logging then?

Ah yes, that works, I've updated the pull request. I still find it a bit confusing to use `setup_logging()` and `basicConfig()` at the same time. Also changing log levels back and forth is not the nicest solution. But it works and I don't see a really good solution without touching aw_core or not using `setup_logging()`. I guess to fix this issue, this should be good enough.

Let me know if you think have any other suggestion!